driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -378,6 +382,17 @@ private boolean addConnectionIfUnderMaximum() {
         }
     }
 
+    private PooledConnection tryResurrectFromTrash() {
+        long now = System.currentTimeMillis();
+        for (PooledConnection connection : trash)
+            if (connection.maxIdleTime >= now && connection.state.compareAndSet(TRASHED, OPEN)) {
```

Comment:
The change looks great, thanks!  I'll update / add integration test where appropriate to handle that scenario.  There are a few other tests I wanted to add too, so I'll commit some more today.